src/aspire/utils/rotation.py

```diff
@@ -0,0 +1,300 @@
+"""
+Define a Rotation Class for customized rotation operations used by ASPIRE.
+"""
+
+import numpy as np
+from numpy.linalg import norm
+from scipy.linalg import svd
+from scipy.spatial.transform import Rotation as sp_rot
+
+from aspire.utils import ensure
+from aspire.utils.random import Random
+
+
+class Rotation:
+    def __init__(
+        self, num_rots, seed=None, angles=None, matrices=None, dtype=np.float32
+    ):
+        """
+         Initialize a Rotation object
+
+         :param num_rots: Total number of rotation sets
+         :param seed: Seed for initializing random rotations.
+             If None, the random generator is not re-seeded.
+         :param angles: Euler angles (in radian) to generate rotation matrices.
+             If None, Predefined rotation matrices or uniformly
+             distributed angles will be used.
+        :param matrices: Rotation matrices to initialize Rotation object.
+             If None, Predefined or uniformly distributed angles
+             will be used.
+         :param dtype: data type for angles and rotation matrices
+        """
+        self.dtype = np.dtype(dtype)
+        self.num_rots = None
```

Comment:
Why do we need to store `num_rots`, `angles`, and `shape` separately? They can all be obtained from `matrices`.

Reply:
When we don't specify angles and matrices such as the case of using random angles, we need num_rots to set the number of rotation matrices.